pkg/services/irma/signer.go

```diff
@@ -0,0 +1,188 @@
+ */
+
+package irma
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/mdp/qrterminal/v3"
+	"github.com/nuts-foundation/nuts-auth/logging"
+	"github.com/nuts-foundation/nuts-auth/pkg/contract"
+	"github.com/nuts-foundation/nuts-auth/pkg/services"
+	"github.com/pkg/errors"
+	irmago "github.com/privacybydesign/irmago"
+	"github.com/privacybydesign/irmago/server"
+)
+
+// SessionPtr should be made private when v0 is removed
+type SessionPtr struct {
+	ID         string
+	QrCodeInfo irmago.Qr `json:"sessionPtr"`
+}
+
+// SessionID returns the SessionID of the SessionPtr
+func (s SessionPtr) SessionID() string {
+	return s.ID
+}
+
+// Payload renders the IrmaQRCode as json according to irmago.Qr
+func (s SessionPtr) Payload() []byte {
+	jsonResult, _ := json.Marshal(s.QrCodeInfo)
+	return jsonResult
+}
+
+// MarshalJSON marshals a custom session pointer json object for the IRMA means.
+func (s SessionPtr) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		QrCodeInfo irmago.Qr `json:"clientPtr"`
+		ID         string    `json:"sessionID"`
+	}{QrCodeInfo: s.QrCodeInfo, ID: s.ID})
+}
+
+// NutsIrmaSignedContract is the type of proof used in an Irma VP
+const NutsIrmaSignedContract = "NutsIrmaSignedContract"
+
+// StartSigningSession accepts a rawContractText and creates an IRMA signing session.
+func (v Service) StartSigningSession(rawContractText string) (contract.SessionPointer, error) {
+	// Put the template in an IRMA envelope
+	signatureRequest := irmago.NewSignatureRequest(rawContractText)
+	schemeManager := v.IrmaServiceConfig.IrmaSchemeManager
+
+	c, err := contract.ParseContractString(rawContractText, v.ContractTemplates)
+	if err != nil {
+		return nil, err
+	}
+
+	var attributes irmago.AttributeCon
+	for _, att := range c.Template.SignerAttributes {
+		// Checks if attribute name start with a dot, if so, add the configured scheme manager.
+		if strings.Index(att, ".") == 0 {
+			att = fmt.Sprintf("%s%s", schemeManager, att)
+		}
+		attributes = append(attributes, irmago.NewAttributeRequest(att))
+	}
+	signatureRequest.Disclose = irmago.AttributeConDisCon{
+		irmago.AttributeDisCon{
+			attributes,
+		},
+	}
+
+	// Start an IRMA session
+	sessionPointer, token, err := v.IrmaSessionHandler.StartSession(signatureRequest, func(result *server.SessionResult) {
+		logging.Log().Debugf("session done, result: %s", server.ToJson(result))
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error while creating session: %w", err)
+	}
+	logging.Log().Debugf("session created with token: %s", token)
+
+	// Return the sessionPointer and sessionId
+	challenge := SessionPtr{
+		ID:         token,
+		QrCodeInfo: *sessionPointer,
+	}
+	jsonResult := challenge.Payload()
+	printQrCode(string(jsonResult))
+
+	return challenge, nil
+}
+
+// SigningSessionStatus returns the current status of a certain session.
+// It returns nil if the session is not found
```

Comment:
The dummy implementation returns an error. Implementations MUST have the same behaviour to adhere to the interface's contract.

Reply:
it was just the comment that was wrong